include/engine/geospatial_query.hpp

```diff
@@ -475,27 +475,21 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
                             forward_durations.begin() + data.fwd_segment_position,
                             EdgeDuration{0});
 
-        // EdgeDistance forward_distance_offset = accumulateDistance(forward_geometry.begin(),
-        // forward_durations.begin() + data.fwd_segment_position, EdgeDistance{0}, datafacade);
-
         EdgeDistance forward_distance_offset = 0;
-        if ((forward_geometry.begin() + data.fwd_segment_position) != forward_geometry.begin())
+        for (auto current = forward_geometry.begin() + 1;
+             current < forward_geometry.begin() + data.fwd_segment_position + 1;
+             ++current)
         {
-            for (auto current = forward_geometry.begin() + 1;
-                 current != forward_geometry.end() - data.fwd_segment_position;
-                 ++current)
-            {
-                forward_distance_offset += util::coordinate_calculation::haversineDistance(
-                    datafacade.GetCoordinateOfNode(*std::prev(current)),
-                    datafacade.GetCoordinateOfNode(*current));
-            }
+            forward_distance_offset += util::coordinate_calculation::haversineDistance(
+                datafacade.GetCoordinateOfNode(*std::prev(current)),
+                datafacade.GetCoordinateOfNode(*current));
         }
 
         EdgeWeight forward_weight = forward_weights[data.fwd_segment_position];
         EdgeDuration forward_duration = forward_durations[data.fwd_segment_position];
         EdgeDistance forward_distance = util::coordinate_calculation::haversineDistance(
             datafacade.GetCoordinateOfNode(forward_geometry(data.fwd_segment_position)),
-            point_on_segment);
+            datafacade.GetCoordinateOfNode(forward_geometry(data.fwd_segment_position+1)));
```

Comment:
@chaupow what you did here by removing calculating the distance from `data.fwd_segment_position` and `point_on_segment` is what multiplying by the ratio below recreates. `point_on_segment` is the location of the phantom_node.

Reply:
Ok, I lacked content on what the point on segment is. But if it is the phantom node, I don't think this should be here? Looking at `forward_duration ` it looks like `forward_distance` should be the distance of the whole segment? I might be wrong tho. Can you try fixing this by using `point_on_segment` and not using the ratio then?